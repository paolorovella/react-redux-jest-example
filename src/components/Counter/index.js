import React from 'react';
import PropTypes from 'prop-types';

const Counter = ({ label, value }) => <span>{label}: {value}</span>;

Counter.defaultProps = {
  label: 'Counter',
  value: 0,
};

Counter.propTypes = {
  label: PropTypes.string,
  value: PropTypes.number,
};

export default Counter;
