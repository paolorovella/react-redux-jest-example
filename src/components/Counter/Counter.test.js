import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '.';

it('<Counter /> renders without crashing', () => {
  const tree = renderer.create(
    <Counter />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Counter /> renders with label as prop', () => {
  const tree = renderer.create(
    <Counter label="Counter" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Counter /> renders with value as prop', () => {
  const tree = renderer.create(
    <Counter value={30} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
