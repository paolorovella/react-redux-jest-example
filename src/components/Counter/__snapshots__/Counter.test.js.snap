// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders with value as prop 1`] = `
<div>
  <div
    className="sc-htpNat iAWPfp"
  >
    <button
      className="sc-bdVaJa bVdEId"
      label="Decrement"
      onClick={[Function]}
      type="danger"
    >
      Decrement
    </button>
    <button
      className="sc-bdVaJa hjTpqZ"
      label="Increment"
      onClick={[Function]}
      type="success"
    >
      Increment
    </button>
  </div>
  <div
    className="sc-bxivhb iPRuUZ"
  >
    <span>
      Counter: 
      30
    </span>
  </div>
</div>
`;

exports[`<Counter /> renders without crashing 1`] = `
<div>
  <div
    className="sc-htpNat iAWPfp"
  >
    <button
      className="sc-bdVaJa bVdEId"
      label="Decrement"
      onClick={[Function]}
      type="danger"
    >
      Decrement
    </button>
    <button
      className="sc-bdVaJa hjTpqZ"
      label="Increment"
      onClick={[Function]}
      type="success"
    >
      Increment
    </button>
  </div>
  <div
    className="sc-bxivhb iPRuUZ"
  >
    <span>
      Counter: 
      0
    </span>
  </div>
</div>
`;
